examples/pytorch/question-answering/sparseml_utils.py

```diff
@@ -119,3 +167,42 @@ def export_model(model, dataloader, output_dir):
         sample_input = (sample_batch["input_ids"], sample_batch["attention_mask"], sample_batch["token_type_ids"])
         exporter.export_onnx(sample_batch=sample_input, convert_qat=True)
         break
+
+
+def preprocess_state_dict(pretrained_model_name_or_path):
+    """
+    Restore original parameter names that were changed by QAT process
+    """
+    state_dict = None
+    if pretrained_model_name_or_path is not None:
+        pretrained_model_name_or_path = str(pretrained_model_name_or_path)
+        if os.path.isdir(pretrained_model_name_or_path):
+            if os.path.isfile(os.path.join(pretrained_model_name_or_path, WEIGHTS_NAME)):
+                archive_file = os.path.join(pretrained_model_name_or_path, WEIGHTS_NAME)
+                state_dict = torch.load(archive_file, map_location="cpu")
+                removed_keys = [
+                    key
+                    for key in state_dict
+                    if key.startswith("bert.encoder.layer.")
```

Comment:
can we safely lift this restriction so it can work with other models?

Reply:
removed the check related to bert, also add a condition that we're under a QAT recipe for this to be applied.